datafusion/Cargo.toml

```diff
@@ -77,6 +77,7 @@ rand = "0.8"
 avro-rs = { version = "0.13", features = ["snappy"], optional = true }
 num-traits = { version = "0.2", optional = true }
 pyo3 = { version = "0.14", optional = true }
+statrs = "0.15"
```

Comment:
I would prefer to avoid more dependencies for datafusion if possible (e.g. so we don't have to deal with https://github.com/apache/arrow-datafusion/issues/1498)


It seems to me like a factorial implementation will not change much and the code from statrs is fairly simple;

https://docs.rs/statrs/0.15.0/src/statrs/function/factorial.rs.html#92-108

Reply:
agree with this.

Reply:
thanks for the suggestion, I removed the dependency